src/main/java/io/openliberty/tools/common/plugins/util/PrepareFeatureUtil.java

```diff
@@ -0,0 +1,416 @@
+	public static final String OPEN_LIBERTY_GROUP_ID = "io.openliberty.features";
+	public static final String INSTALL_MAP_ARTIFACT_ID = "install-map";
+	public static final String FEATURES_JSON_ARTIFACT_ID = "features";
+	private static final String MIN_USER_FEATURE_VERSION = "21.0.0.6";
+
+	private File installJarFile;
+
+	public PrepareFeatureUtil(File installDirectory, String openLibertyVersion)
+			throws PluginScenarioException, PluginExecutionException {
+		this.installDirectory = installDirectory;
+		this.openLibertyVersion = openLibertyVersion;
+		installJarFile = loadInstallJarFile(installDirectory);
+
+		// check if the openliberty kernel meets min required version 21.0.0.6
+		DefaultArtifactVersion minVersion = new DefaultArtifactVersion(MIN_USER_FEATURE_VERSION);
+		DefaultArtifactVersion version = new DefaultArtifactVersion(openLibertyVersion);
+
+		if (version.compareTo(minVersion) < 0) {
+			throw new PluginScenarioException(
+					"To install user feature, openliberty version should be greater than 21.0.0.6");
+		}
+		if (installJarFile == null) {
+			throw new PluginScenarioException("Install map jar not found.");
+		}
+	}
+
+	public void prepareFeatures(List<String> featureBOMs) throws PluginExecutionException {
+		for (String BOMCoordinate : featureBOMs) {
+			String[] coord = BOMCoordinate.split(":");
+			String groupId = coord[0];
+			String artifactId = coord[1];
+			String version = coord[2];
+			prepareFeature(groupId, artifactId, version);
+		}
+
+	}
+
+	/**
+	 * generate JSON if not already present at the desired location
+	 * 
+	 * @param groupId
+	 * @param artifactId
+	 * @param version
+	 * @throws PluginExecutionException
+	 */
+	private void prepareFeature(String groupId, String artifactId, String version) throws PluginExecutionException {
+		File json = null;
+		try {
+			json = downloadArtifact(groupId, FEATURES_JSON_ARTIFACT_ID, "json", version);
+		} catch (PluginExecutionException e) {
+			debug(e);
+			info(String.format("Cannot find json for coordinate %s:%s:%s in connected repositories", groupId,
+					artifactId, version));
+		}
+		if (json != null) {
+			info("The features.json already exists at the following location: " + json);
+		} else {
+			try {
+				File additionalBOM = downloadArtifact(groupId, artifactId, "pom", version);
+				String repoLocation = parseRepositoryLocation(additionalBOM, groupId, artifactId, "pom", version);
+				String targetJsonFile = createArtifactFilePath(repoLocation, groupId, FEATURES_JSON_ARTIFACT_ID, "json",
+						version);
+				Map<File, String> esaFiles = downloadArtifactsFromBOM(additionalBOM);
+				File generatedJson = generateJson(targetJsonFile, esaFiles);
+				if (generatedJson.exists()) {
+					info("Json has been generated at the following location: " + generatedJson);
+				}
+			} catch (PluginExecutionException e) {
+				warn("unable to find BOM file for given groupId " + groupId);
+				error("expecting BOM file artifact to be in repository: " + e.getMessage());
```

Comment:
All the info/warn/error messages in this method should be reworked. They need to start with capital letters, be complete sentences and contain enough information to be useful.